src/driver.toit

```diff
@@ -71,19 +119,46 @@ class Driver:
     waiters_.add latch
     return latch.get
 
-  run:
+
+  /**
+  Starts the message receiver task.
+
+  Function returns a latch.  When the driver is being used manually (see
+    `--auto-start` in the constructor), the latch is set when the task has fully
+    started. This is useful to ensure the task has fully started before sending
+    CFG messages that would otherwise block until a ACK/NAK was received.
+  */
+  run -> monitor.Latch:
     assert: not runner_
     adapter_.flush
-    start-periodic-nav-packets_
+    start-latch := monitor.Latch
+
+    // Start the message parser task to parse messages as they arrive.
     runner_ = task::
+      start-latch.set true
       while true:
         message := adapter_.next-message
-        if message is ubx-message.NavStatus:
+        //logger_.debug  "Received: $message"
+
+        if message is ubx-message.AckAck:
+          // Message is an ACK-ACK - positive response to a CFG message.
```

Comment:
I would put that comment into the `process-ack-ack-message_`. Same for the nack one.